### ISSUE TITLE
First try

### ISSUE THREAD

**Issue by marxelo** [marxelo/Professional-Scrum-Developer-I-PSD-I-Practice-Tests-Exams-Questions-Answers#1]
### When can Product Backlog Refinement occur?

- [ ] Only during Sprint Planning.
- [x] Anytime during the Sprint.
- [ ] Only during Refinement meetings planned by the Product Owner.
- [ ] Before Sprint Planning.

### What would NOT be considered Refactoring?

- [ ] Reordering method parameters to improve readability.
- [ ] Extracting interfaces.
- [ ] Renaming things to be more logical.
- [ ] Changing external interfaces or APIS.
- [ ] Extracting methods.

### Your Scrum Team is one of seven teams working on a Software Product. All teams use the same Version Control System. Which is the best approach to deliver a high-quality Increment?

- [ ] Developers should perform a combination of local and private builds.
- [ ] Each team's automated build is integrated toward the end of the Sprint.
- [ ] There is one automated and integrated build for all seven teams.
- [ ] Each team should have its own automated build.

### Who creates documentation included with an Increment?

- [ ] The Developers.
- [ ] Increments do not need documentation.
- [ ] The Product Owner.
- [ ] Technical Writers.
- [ ] The Scrum Master.

### What is a merge in a Version Control System?

- [ ] Copying a portion of a code base to isolate it from the original codebase.
- [ ] Identifying a particular codebase as ready for distribution.
- [ ] Triggering a Deployment into Production.
- [ ] Combining two or more versions of code into a single codebase.

### Which three of the following are feedback loops in Scrum?

- [ ] Daily Scrum.
- [ ] Release Planning.
- [ ] Sprint Review.
- [ ] Refinement Meeting.
- [ ] Sprint Retrospective.

### Which four types of tests can be included in an automated test harness?

- [ ] Performance.
- [ ] Exploratory.
- [ ] Unit.
- [ ] Functional.
- [ ] Manual regression.
- [ ] Integration.

### Which concept is described by the Last Responsible Moment?

- [ ] Making decisions as soon as possible to close feedback loops as soon as possible.
- [ ] Discover decisions to be made as soon as possible but postpone deciding to the latest reasonable moment.
- [ ] The last moment in a Sprint when code changes are allowed, after this only stabilization work should be conducted.
- [ ] Opening a learning window to validate hypotheses and create learning.
- [ ] The last moment a Developer is responsible for quality, after this the Tester is responsible.

### Choose four desirable characteristics of a Unit Test.

- [ ] Includes exercising the persistence layer.
- [ ] Makes assertions about only one logical concept.
- [ ] Independent of others.
- [ ] Test code is as small as possible.
- [ ] Execution is fast.

### Why does a test written using TDD (Test Driven Development) initially fail?

- [ ] Because the test has not been refactored.
- [ ] Because it has to be put into an automated test harness to be run.
- [ ] Because the tests are checked in before the Product code exists.
- [ ] Because the Product code to satisfy the test does not yet exist.

### When do the Developers show their work to the Product Owner?

- [ ] Whenever the Product Owner asks.
- [ ] During the Sprint Review.
- [ ] Anytime the Developers need feedback from the Product Owner.
- [ ] All of the above.

### What are possible benefits from practicing Test Driven Development (TDD)?

- [ ] It is a great way for Testers to contribute early in the Sprint as they can start creating all the tests from the beginning on.
- [ ] It makes Integration Tests obsolete and by doing so saves a lot of time.
- [ ] It helps break down complex problems into smaller ones that are much easier to be tackled.
- [ ] It increases collaboration between Developers as it requires Pair Programming.
- [ ] It might lead to better code as Refactoring is part of the TDD cycle.
- [ ] It helps to identify gaps in understanding the desired behavior.

### Who decides the System Architecture of a Product developed using Scrum?

- [ ] The Architect chosen by the Scrum Team.
- [ ] The Software Architect assigned to the Scrum Team.
- [ ] The Developers with input from the Scrum Team and others.
- [ ] The Chief Architect.

### When using Continuous Integration, how often should the build be executed?

- [ ] Once per hour.
- [ ] Whenever new or changed code is checked into version control.
- [ ] Once per day.
- [ ] Before the end of the Sprint.
- [ ] Whenever new tests are created or uncertainty arises about whether old tests will pass.

### Which of the following are advantages of Continuous Integration?

- [ ] Readability of code is improved.
- [ ] Know immediately how a change affected the Product.
- [ ] The feature-branch is generally kept in a buildable state.
- [ ] Reduce effort and risk when integrating changes.

### Who determines how work is performed during the Sprint?

- [ ] The Scrum Master.
- [ ] The Scrum Team.
- [ ] Team Manager.
- [ ] Subject matter experts.
- [ ] The Developers.

### Who creates tests on a Scrum Team?

- [ ] The Product Owner.
- [ ] The Developers.
- [ ] Quality Assurance Specialists.
- [ ] The Scrum Master.

### What is Test First Development (TFD)?

- [ ] Creating tests before satisfying them.
- [ ] Testing existing code before adding more code to it.
- [ ] Having the Tester in the Scrum Team write the test plans before coding.
- [ ] The continuous restructuring of Software to retain flexibility.

### Which is the most reliable form of technical documentation?

- [ ] A spreadsheet of passing manual tests.
- [ ] The Developer's whiteboard.
- [ ] UML model.
- [ ] Release notes.
- [ ] A help file.
- [ ] A passing test harness with clear naming and vocabulary.

### While developing new functionality, you find a bug that has already been delivered to the customer. What do you do?

- [ ] Revise the tests so that the bug no longer appears on the bug report.
- [ ] Fix the bug.
- [ ] Talk to the Product Owner.
- [ ] Stub out the code that caused the bug so it no longer occurs.

### Who has the final decision about the order of items in the Product Backlog?

- [ ] The Scrum Master.
- [ ] The Stakeholders.
- [ ] The Scrum Team.
- [ ] The Developers.
- [ ] The Product Owner.

### What are two differences between Unit Tests and Integration Tests?

- [ ] A Unit Test isolates a specific system behavior.
- [ ] An Integration Test usually focuses on the integration of two or more units.
- [ ] A Unit Test is automated.
- [ ] An Integration Test runs overnight.
- [ ] A Unit Test only runs on a Developer's workstation.

### How much time must a Product Owner spend with the Developers?

- [ ] 100%.
- [ ] Enough so that the Product Owner is not surprised by the value delivered by the Increment.
- [ ] 40%, or more if the Stakeholders agree.
- [ ] Any amount of time the Developers ask the Product Owner to be present.

### What are the typical roles when practicing Pair Programming?

- [ ] Product Owner and Developer.
- [ ] Business Analyst and Developer.
- [ ] Tester and Developer.
- [ ] Driver and Navigator.
- [ ] Frontend and Backend.

### True or False: When multiple teams work together on the same Product, each team should maintain a separate Product Backlog.

- [ ] True.
- [ ] False.

### What are three of the best ways to address non-functional requirements?

- [ ] Scrum is for functional, front-end development only.
- [ ] Important, recurring non-functional requirements can be added to the Definition of Done.
- [ ] Discuss them during a risk mitigation phase before development.
- [ ] Include them in the Product Backlog.
- [ ] Specific expectations can be used as Acceptance Criteria to specific Product Backlog Items.
- [ ] Before the release, they should be tested and validated in a hardening Sprint.

### What is the primary purpose of Refactoring?

- [ ] Ensuring that all factors are constantly aligned.
- [ ] Removing all bugs that were found during Regression Tests.
- [ ] Making sure that the code is readable and maintainable.
- [ ] Creating better technical documentation.

### Which are four attributes of a good bug report?

- [ ] Expected results and observed results.
- [ ] Includes build or version number where bug was found.
- [ ] Includes code for a proposed fix.
- [ ] Provides simple and repeatable reproduction steps.
- [ ] Screenshots or other pictures of the bug in action.
- [ ] Explains some new system functionality desired.

### The practice of decomposing a requirement into failing tests is called:

- [ ] Regression Testing.
- [ ] Object oriented requirements definition.
- [ ] Acceptance Test Driven Development.
- [ ] Behavior Driven Development.

### What is a mock object?

- [ ] A test object that mimics the behavior of a dependency in the system under test.
- [ ] A mock helps you create a build script.
- [ ] Mocks, stubs, dummies, fakes, and shims are all the same.
- [ ] A mock is a way to initialize the database for testing.

### Which four are benefits of Test Driven Development?

- [ ] It improves quality and reduces bugs.
- [ ] It reduces the cost of maintenance over time.
- [ ] It promotes good design and separation of concerns.
- [ ] It ensures no defects are present in the code.
- [ ] It causes you to construct a test harness that can be automated.

### Which are three attributes of a bad bug report?

- [ ] Vague statements or untested assumptions.
- [ ] Generic titles.
- [ ] Simple and repeatable reproduction steps.
- [ ] Assigning blame.
- [ ] One bug per report.

### When using Scrum, can a Scrum Team use Continuous Delivery?

- [ ] Yes, there is nothing in Scrum that conflicts with Continuous Delivery.
- [ ] No, because the Product Owner may not be available each time a PBI is done and the Product Owner has to decide if it should be released.
- [ ] No, because the increment has to be approved at the Sprint Review before it can be released, and a 2-week Sprint would not be considered continuous.
- [ ] No, because Stakeholders may have already seen the new features and the Sprint Review will be meaningless.
- [ ] No, because before releasing an Increment it first has to fulfill the Definition of Done.

### You are asked to refactor part of the codebase for Application X. When you are done, all Unit Tests pass with 50% Code Coverage. What can you infer from this?

- [ ] At least 50% of Application X functions correctly.
- [ ] At most 50% of Application X functions correctly.
- [ ] There are no bugs present in Application X.
- [ ] You did not break any existing Unit Tests.

### What are two ways that regulatory compliance issues are dealt with in Scrum?

- [ ] They are addressed by a separate team who is responsible for compliance issues.
- [ ] They are addressed along with functional development of the Product.
- [ ] They are discussed, determined, and documented before the actual feature development Sprints.
- [ ] They are added to the Product Backlog and addressed in early Sprints, while always requiring at least some business functionality. no matter how small.

### True or False: User Stories are required in the Product Backlog.

- [ ] True.
- [ ] False.

### In Software Development, DRY refers to:

- [ ] Code with low Cyclomatic Complexity.
- [ ] Code with minimal duplication.
- [ ] Code that has not been peer reviewed.
- [ ] Code that has been peer reviewed.

### Which statements are true when multiple Scrum Teams work on a Software Product at the same time?

- [ ] All Scrum Teams work in their own version control branch.
- [ ] Code is merged at the Scrum of Scrums.
- [ ] The Scrum Teams must integrate their work before the end of the Sprint.
- [ ] The Scrum Teams coordinate their work to deliver a single Increment.
- [ ] Each Scrum Team should have a different Product Owner.

### Which answer best describes Behavior Driven Development (BDD)?

- [ ] A style of Test Driven Development focusing on user and system interactions.
- [ ] A development style that accounts for leadership style among team members.
- [ ] A way to organize Unit Tests based on class and method structures.
- [ ] A technique for maintaining Regression Test harnesses.

### In what ways do Developers contribute to refining the Product Backlog?

- [ ] They ask questions in order to clarify the intent of Product Backlog Items.
- [ ] They do not. The Developers are only responsible for prioritizing technical work.
- [ ] They do not. The Scrum Master and the Product Owner are responsible for Product Backlog Refinement.
- [ ] They give input on technical dependencies.
- [ ] They may update estimates for Product Backlog Items.

### When should the Developers create their first automated build?

- [ ] Just before the Product is released.
- [ ] Before writing the first line of code.
- [ ] When the Product Owner asks for a build.
- [ ] Just before the end of the Sprint.
- [ ] As soon as there is code in the Version Control System.

### Which types of tests can be automated?

- [ ] Exploratory.
- [ ] Performance.
- [ ] Unit.
- [ ] Smoke.
- [ ] Functional.
- [ ] Integration.

### How much work is required of the Developers to complete a Product Backlog Item selected during the Sprint Planning?

- [ ] A proportional amount of time on analysis, design, development, and testing.
- [ ] All development work and at least some testing.
- [ ] As much as they can fit into the Sprint, with remaining work deferred to the next Sprint.
- [ ] As much as s required to meet the Scrum Team's Definition of Done.

### Why might Developers choose to do Pair Programming?

- [ ] Information sharing and learning.
- [ ] Improving Code Quality.
- [ ] Efficiency.
- [ ] It can be a fun way to work.
- [ ] All of the above.

### What is the role of Modeling in Scrum Teams?

- [ ] Models are not used by agile teams.
- [ ] Models are maintained along with the Software as it emerges.
- [ ] Models are assembly instructions for the Developers.
- [ ] Modeling may be useful to increase shared understanding.

### Which of the following are DevOps Practices?

- [ ] Continuous Integration / Continuous Deployment / Continuous Delivery.
- [ ] Blue-Green Deployment.
- [ ] Hypothesis Driven Development.
- [ ] Vertical Teams.
- [ ] Blameless Postmortem.
- [ ] All of the above.

### While practicing Test Driven Development, what is done after the test fails?

- [ ] Write the minimum amount of Product code to satisfy the test.
- [ ] Refactor the test so the code passes.
- [ ] Run it again to make sure it really fails.
- [ ] Implement the required functionality.
- [ ] Meet with the Business Analyst to ensure that the test is correct.

### What tactic should a Scrum Master use to divide a group of 100 people into multiple Scrum Teams?

- [ ] Create teams based on their skills across multiple layers (such as database, Ul, etc.).
- [ ] Ask the people to divide themselves into teams.
- [ ] Ask the Product Owner to assign the people to teams.

### Developers are blocked by an impediment in the middle of the Sprint. The impediment is outside the Developer's control. What should they do?

- [ ] Stop using Scrum until the impediment is resolved.
- [ ] Complete the work that can be done and complete the remainder during the hardening Sprint.
- [ ] Immediately raise the issue to the Scrum Master.
- [ ] Cancel the Sprint.
- [ ] Drop the Product Backlog Items affected by the impediment from the Sprint Plan.

### What is an Integration Test?

- [ ] A test of the user interface.
- [ ] A test runs during a Continuous Integration build.
- [ ] A test of a single unit of functionality.
- [ ] A test of multiple units of functionality.

### Which of the following describes the focus of the first way of DevOps?

- [ ] The first set of practices a team should apply before moving to the second way.
- [ ] Using automated build and release pipelines.
- [ ] To deliver value earlier and more frequently.
- [ ] A tool-focused way of introducing DevOps, compared to a mindset way (second way) and organizational structure (third way).
- [ ] A culture of continuous experimentation and learning.

### Which of the following is true about the Definition of Done?

- [ ] It might be a subject of discussion during Sprint Retrospective.
- [ ] It is the sole responsibility of the Developers to define it.
- [ ] It is synonymous with Acceptance Criteria.
- [ ] It can only be extended; nothing can be removed.
- [ ] It defines a state when the entire Increment is releasable.

### At Sprint Planning, the Scrum Team has NO clear standard to meet for releasable Software. What should the Scrum Team do?

- [ ] Ask the Product Owner to specify a Definition of Done.
- [ ] Create a unique completion checklist for each item in the Sprint.
- [ ] Specify a shared Definition of Done.
- [ ] Ask the Scrum Master what they should do.

### Based on "Sprint 7 Burndown Chart" would you do anything different in Sprint 8?

![Burndown](images/burndown.jpg)

- [ ] The Scrum Master adds additional Developers for Sprint 8.
- [ ] The Developers carry over incomplete Sprint Backlog items from Sprint 7 to Sprint 8 and monitor the Sprint 8 burn-down chart. As soon as deviation from trends is detected, the Developers work with the Product Owner to negotiate remaining work.
- [ ] There is nothing wrong The Developers will present all Product Backlog Items selected for Sprint 7 at the end of the Sprint.
- [ ] The Developers may forecast less overall work in Sprint 8.
- [ ] Stakeholders will encourage the Scrum Team to estimate better during the Sprint 8 Planning Meeting.
- [ ] The Developers put incomplete Product Backlog Items back into the Product Backlog for re-ordering.
- [ ] The Product Owner may ask the Developers to complete the unfinished Product Backlog Items from Sprint 7 in Sprint 8.

### True or False: Database design must be complete before coding starts to ensure a solid foundation?

- [ ] True.
- [ ] False.

### The Daily Scrum event happens every day. What would be three concerns if the frequency were to be lowered to every two to three days?

- [ ] Opportunities to inspect and adapt in the Sprint Backlog are lost.
- [ ] The Scrum Master loses the ability to update the Gantt Chart properly.
- [ ] Too much work is spent updating the Scrum Board before meeting.
- [ ] Sprint Plan may become inaccurate.
- [ ] Impediments are raised and resolved more slowly.
- [ ] The Product Owner cannot accurately report to the Stakeholders.

### Which of the following are required by Scrum? (choose all of those who apply)

- [ ] Release Burnup Chart.
- [ ] Burndown Chart.
- [ ] Unit Tests.
- [ ] Critical Path Analysis.
- [ ] Refactoring.
- [ ] Build automation.
- [ ] None of the above.

### Who is responsible for creation of the Definition of Done?

- [ ] The Scrum Master.
- [ ] The Scrum Team.
- [ ] The Development Team.
- [ ] The Product Owner.

### Pair Programming is implemented by?

- [ ] Frontend and Backend Developers.
- [ ] Tester and Developer.
- [ ] Developer and Scrum Master.
- [ ] Two persons working on the same PBI.

### Which of the following best describes Emergent Architecture?

- [ ] Starting development of a Software Product requires a clear understanding of the underlying Software Architecture and Emergent Architecture describes the process of creating this initial architecture.
- [ ] The Software Architecture emerges solely decisions the Developers make from a technical perspective.
- [ ] In Scrum there is no Architecture-Role therefore architecture emerges naturally.
- [ ] The desire to make decisions easier to change in the future and find the best possible point in time to make decisions.
- [ ] Enterprise Architects must be involved to create the foundation of each Software Product.

### Which are two benefits of establishing naming conventions for code? (choose best two answers)

- [ ] To make it easy to distinguish between different Software Products.
- [ ] To make the code more readable.
- [ ] To communicate the identity of the Developer who worked on the code.
- [ ] To reduce friction in a shared codebase.
- [ ] To ensure that abandoned functions are not created.

### Does delivering changes frequently and directly into Production help to reduce risk? (choose two)

- [ ] Yes, because the changes you make are much smaller and it is easier to fix problems.
- [ ] No, because each release means a risk to break something so releasing more frequently would increase the risk.
- [ ] Yes, because frequent release encourage the Developers to automate the release process.
- [ ] Yes, because you get earlier feedback and can learn faster.
- [ ] No, the only way to tackle these risks is by extensive risk management.
- [ ] No, because releasing needs extensive testing which cannot be conducted frequently.

### True or False: Best Practices are recommended to solve complex problems.

- [ ] True.
- [ ] False.

### When is Performance Testing most effectively performed?

- [ ] Often, throughout development of the Software.
- [ ] Just before deploying to Production.
- [ ] After coding is complete.
- [ ] In Production.

### Which three of the following criteria are most helpful as part of a Scrum Team's Definition of Done? (choose best three answers)

- [ ] The Product is released at the end of every Sprint.
- [ ] Acceptance Tests pass.
- [ ] Code Review is done.
- [ ] Regression Tests pass.
- [ ] No impediments exist.

### A team has expressed requirements as a set of failing Acceptance Tests. What are two benefits? (choose best two answers)

- [ ] Improves quality in the requirement itself.
- [ ] Clear Acceptance Criteria for each feature.
- [ ] Using a code generation tool, the solution can be generated from the requirements model.
- [ ] Promotes the use of DRY principle.
- [ ] Tracking of competencies.

### Which is LEAST useful when measuring Code Maintainability?

- [ ] Function Points.
- [ ] Cyclomatic Complexity.
- [ ] Depth of Inheritance.

### Which two criteria are useful in deciding if something should be documented every Sprint? (choose best two answers)

- [ ] The Scrum Master requires it.
- [ ] It is required by the Definition of Done.
- [ ] The documentation is used to enhance and maintain the Software.
- [ ] The Software tool being used requires it.
- [ ] It has always been documented in the past.

### What factor should be considered when establishing the Sprint length?

- [ ] The need of the team to learn on doing work and measuring results.
- [ ] The frequency at which team formation can be changed.
- [ ] The organization release schedule.
- [ ] The organization has mandated similar length Sprints.

### What are some disadvantages of Code Coverages as a measurement for how well a system or Product is tested? (choose best three answers)

- [ ] It only provides insights for programmers.
- [ ] It does not ensure that the most important or highest risk areas of the code are being exercised by the tests.
- [ ] It could create incentives to write tests that simply increase Code Coverage, rather than tests that find bugs without increasing coverage.
- [ ] It is too complicated to explain to Management.
- [ ] Developers could stop adding more valuable tests once the target coverage is achieved.

### Which of the following are quality goals in Application Architecture? (choose best two answers)

- [ ] Build.
- [ ] Security.
- [ ] Design Pattern selection.
- [ ] Scalability.

### True or False: Stakeholders can be included in Product Backlog Refinement?

- [ ] True.
- [ ] False.

### What are two good ways for the Development Team to make a non-functional requirements visible?

- [ ] Put them on a separate list on the Scrum board, available for all to see.
- [ ] Add them to the Product Backlog and keep the Product Owner posted on the expected effort.
- [ ] Run the Integration and Regression Tests before the end of the Sprint, and capture the open work for the Sprint Backlog of the next Sprint.
- [ ] Add them to the Definition of Done so the work is taken care of every Sprint.

### What activities would a Product Owner typically undertake in the phase between the end of the current Sprint and the start of the next Sprint? (choose the best answer)

- [ ] There are no such activities. The next Sprint starts immediately after the current Sprint.
- [ ] Work with the Quality Assurance departments on the Increment of the current Sprint.
- [ ] Refine the Product Backlog.
- [ ] Update the project plan with Stakeholders.

### You are the Scrum Master on a newly formed Scrum Team. Which two of the following activities would probably help the team in starting up? (choose the best two answers)

- [ ] Ensure the Scrum Team members have compatible personalities.
- [ ] Introduce a bonus system for the top performers in the team.
- [ ] Ensure the team understands they need a Definition of Done.
- [ ] Have the development managers for each Development Team member introduce their direct reports and go over their responsibilities on the Scrum Team.
- [ ] Ask the Product Owner to discuss the Product, its vision, history, goals, and context, as well as answer questions.

### Which best describes the Product Backlog? (choose the best answer)

- [ ] It contains all foreseeable tasks and requirements from which the Scrum team can develop and maintain a complete project plan.
- [ ] It is allowed to grow and change as more is learned about the Product and its customers.
- [ ] It is baselined to follow change management processes.
- [ ] It provides just enough information to enable a Scrum team to start the design phase of a Product.

### What are two responsibilities of Testers in a Development Team? (choose the best two answers)

- [ ] Verifying the work of programmers.
- [ ] Scrum has no "Tester" role.
- [ ] Finding bugs.
- [ ] Everyone in the Development Team is responsible for quality.
- [ ] Tracking quality metrics.

### How do you know that a Development Team is cross-functional? (choose the best answer)

- [ ] A few of the Development Team members pair program and do Test Driven Development.
- [ ] Development Team has all the skills to create a potentially releasable increment by the end of every Sprint.
- [ ] Every member of the Development Team is able to perform every task.
- [ ] There are no conflicts within the Development Team.

### For the purpose of transparency, when does Scrum say a new Increment of working Software must be available? (choose the best answer)

- [ ] When the Product Owner asks to create one.
- [ ] At the end of every Sprint.
- [ ] Before the release Sprint.
- [ ] Every 3 Sprints.
- [ ] After the Acceptance Testing phase.

### Which three behaviors demonstrate that a team is self-organizing? (choose the best three answers)

- [ ] The Development Team inviting external people to the Sprint Planning to ask them how to turn a Product Backlog Item into an Increment via a complete and detailed Sprint Backlog.
- [ ] The Development Team members are working within the boundaries of their functional description and nicely handing off work from analyst to Developer to Tester to integration.
- [ ] The Product Owner doesn't need to be at Sprint Retrospectives.
- [ ] Stakeholders walking in at the Daily Scrum to check progress and work with the Scrum Master to optimize the functional scope for the Sprint.
- [ ] Development Team members collaboratively selecting their own work during the Sprint.
- [ ] The Scrum Master is no longer needed.
- [ ] The Development Team has all the skills needed to create a releasable Increment.
- [ ] The Development Team creating their own Sprint Backlog, reflecting all work that is part of the Definition of Done.

### What is the purpose of a Sprint Review? (choose the best answer)

- [ ] To take time to judge the validity of the project.
- [ ] To inspect the Product Increment with the Stakeholders and collect feedback on next steps.
- [ ] To review the Scrum Team's activities and processes during the Sprint.
- [ ] To build team spirit.

### True or False: The Product Owner makes sure the team selects enough from the Product Backlog for a Sprint to satisfy the Stakeholders.

- [ ] True.
- [ ] False.

### Which statement best describes the Sprint Backlog as outcome of the Sprint Planning? (choose the best answer)

- [ ] It is the Development Team's plan for the Sprint.
- [ ] Every item has a designated owner.
- [ ] It is a complete list of all work to be done in a Sprint.
- [ ] Each task is estimated in hours.
- [ ] It is ordered by the Product Owner.

### A Development Team selects a set of Product Backlog Items for a Sprint Backlog with the intent to get the selected items "Done" by the end of the Sprint. Which three phrases best describe the purpose of a Definition of Done? (choose the best three answers)

- [ ] It provides a template for elements that need to be included in the technical documentation.
- [ ] It guides the Development Team in creating a forecast at the Sprint Planning.
- [ ] It controls whether the Developers have performed their tasks.
- [ ] It creates transparency over the work inspected at the Sprint Review.
- [ ] It defines what it takes for an Increment to be ready for release.
- [ ] It tracks the percentage complete of a Product Backlog Item.

### Which output from Sprint Planning provides the Development Team with a target and overarching direction for the Sprint? (choose the best answer)

- [ ] The Sprint Goal.
- [ ] Sprint Review minutes.
- [ ] The Release Plan.
- [ ] The Sprint Backlog.

### The Product Owner determines how many Product Backlog Items the Development Team selects for a Sprint. (choose the best answer)

- [ ] True, but only after confirmation by the resource manager that the Team has enough capacity.
- [ ] False, capacity and commitment are the Project Manager's responsibility.
- [ ] False, the Scrum Master does that.
- [ ] True.
- [ ] True, accordingly to what was committed to the Stakeholders.
- [ ] False.

### Who owns the Sprint Backlog? (choose the best answer)

- [ ] The Scrum Team.
- [ ] The Scrum Master.
- [ ] The Development Team.
- [ ] The Product Owner.

### When is implementation of a Product Backlog Item considered complete? (choose the best answer)

- [ ] When the item has no work remaining in order to be potentially released.
- [ ] When Quality Assurance reports that the item passes all Acceptance Criteria.
- [ ] At the end of the Sprint.
- [ ] When all work in the Sprint Backlog related to the item is finished.

### Which two of the following are true about the Scrum Master role? (choose the best two answers)

- [ ] The Scrum Master teaches the Development Team to keep the Scrum meetings to their time-box.
- [ ] The Scrum Master is responsible for updating the Sprint Burndown.
- [ ] The Scrum Master assigns tasks to Development Team members when they need work.
- [ ] The Scrum Master helps those outside the team interact with the Scrum Team.
- [ ] At the Sprint Review, the Scrum Master identifies what has been "Done" and what has not been "Done".

### Which two of the following are appropriate topics for discussion during a Sprint Retrospective? (choose the best two answers)

- [ ] Documenting Acceptance Criteria for items in the next Sprint. 
- [ ] The order of items in the Product Backlog.
- [ ] Identifying high priority process improvements for the next Sprint.
- [ ] How the team collaborates.

### True or False: Multiple Scrum Teams working on the same project must have the same Sprint start date.

- [ ] True.
- [ ] False.

### When is it most appropriate for a Development Team to change the Definition of Done? (choose the best answer)

- [ ] During Sprint Planning.
- [ ] Prior to starting a new project.
- [ ] Prior to starting a new Sprint.
- [ ] During the Sprint Retrospective.

### The Product Owner is not collaborating with the Development Team during the Sprint. What are two valuable actions for a Scrum Master to take? (choose the best two answers)

- [ ] Nominate a proxy Product Owner.
- [ ] Coach the Product Owner in the values of Scrum and incremental delivery.
- [ ] Stop the Sprint, send the Product Owner to a course and restart.
- [ ] Inform the Product Owner's functional manager.
- [ ] Bring up the problem in the Sprint Retrospective.

### A Scrum Master is working with a Development Team that has members in different physical locations. The Development Team meets in a variety of meeting rooms and has much to do logistically (for example, set up conference calls) before the Daily Scrum. What action should the Scrum Master take? (choose the best answer)

- [ ] Ask the Development Team members to alternate who is responsible for meeting setup.
- [ ] Set up the meeting and tell the Development Team that is how it will be done.
- [ ] Inform Management and ask them to solve it.
- [ ] Allow the Development Team to self-manage and determine for itself what to do.

### Five new Scrum Teams have been created to build one Product. A few of the Developers on one of the Development Teams ask the Scrum Master how to coordinate their work with the other teams. What should the Scrum Master do?

- [ ] Teach the Product Owner to work with the Lead Developers on ordering Product Backlog in a way to avoid too much technical and development overlap during a Sprint.
- [ ] Collect the Sprint tasks from the teams at the end of their Sprint Planning and merge that into a consolidated plan for the entire Sprint.
- [ ] Teach them that it is their responsibility to work with the other teams to create an integrated Increment.
- [ ] Visit the five teams each day to inspect that their Sprint Backlogs are aligned.

### True or False: Scrum is a methodology that tells in detail how to build Software incrementally.

- [ ] True.
- [ ] False.

### In the Sprint Planning meeting, the Product Owner and the Development Team were unable to reach a clear understanding about the highest order Product Backlog Items. Because of this, the Development Team couldn't figure out how many Product Backlog Items it could forecast for the upcoming Sprint. They were able to agree on a Sprint Goal, however. Which of the following two actions should the Scrum Master support? (choose the best two answers)

- [ ] Ask everyone to take as much time as needed to analyze the Product Backlog first, and then reconvene another Sprint Planning meeting.
- [ ] Cancel the Sprint. Send the entire team to an advanced Scrum training and then start a new Sprint.
- [ ] Forecast the most likely Product Backlog Items to meet the goal and create a Sprint Backlog based on a likely initial design and plan. Once the time-box for the Sprint Planning meeting is over, start the Sprint and continue to analyze, decompose, and create additional functionality during the Sprint.
- [ ] Continue the Sprint Planning meeting past its time-box until an adequate number of Product Backlog Items are well enough understood for the Development Team to make a complete forecast. Then start the Sprint.
- [ ] Discuss in the upcoming Sprint Retrospective why this happened and what changes will make it less likely to recur.

### A member of the Development Team takes the Scrum Master aside to express his concerns about data security issues. What should the Scrum Master do? (choose the best answer)

- [ ] Create a Product Backlog Item for security.
- [ ] Ask the person to share the issue with the team as soon as possible.
- [ ] Add security to the Definition of Done.
- [ ] Tell the Product Owner to stop further development of features until the issues are fixed.
- [ ] Go check with the Testers.

### What are two ways that architecture and infrastructure are handled in Scrum? (choose the best two answers)

- [ ] They are implemented along with functional development of the Product.
- [ ] They are built by a separate team through the creation of an architectural runway.
- [ ] They are added to the Product Backlog and addressed in early Sprints, while always requiring at least some business functionality, no matter how small.
- [ ] They are discussed, determined, and documented before the actual feature development Sprints.

### What are three ways Scrum promotes self-organization? (choose three)

- [ ] By preventing Stakeholders from entering the development room.
- [ ] By removing titles for Development Team members.
- [ ] By not allowing documentation.
- [ ] By being a lightweight framework.
- [ ] By the Development Team deciding what work to do in a Sprint.

### True or False: Cross-functional teams are optimized to work on one technical layer of a system only (e.g. GUI, database, middle tier, interfaces).

- [ ] False.
- [ ] True.

### What are three benefits of self-organization? (choose three)

- [ ] Increased accuracy of estimates.
- [ ] Increased rule compliance.
- [ ] Increased self-accountability.
- [ ] Increased creativity.
- [ ] Increased commitment.

### Why does a Development Team need a Sprint Goal?

- [ ] A Sprint Goal ensures that all of the Product Backlog Items selected for the Sprint are implemented.
- [ ] The Development Team is more focused with a common yet specific goal.
- [ ] Sprint Goals are not valuable. Everything is known from the Product Backlog.
- [ ] A Sprint Goal only gives purpose to Sprint 0.

### How should Product Backlog Items be chosen when multiple Scrum Teams work from the same Product Backlog?

- [ ] The Product Owner should provide each team with its own Product Backlog.
- [ ] The Development Teams pull in work in agreement with the Product Owner.
- [ ] Each Scrum Team takes an equal numbers of items.
- [ ] The Scrum Team with the highest velocity pulls Product Backlog Items first.
- [ ] The Product Owner decides.

### Which of the following describe Test Driven Development? (choose two answers)

- [ ] A Software Development technique based on automated tests.
- [ ] A predictable way to develop working, well-organized code.
- [ ] Having Testers in the development process.
- [ ] An Increment and emergent approach to Software Design.
- [ ] Creating a manual test script before writing code.

### When a Continuous Integration build fails, who ideally ensures the build is repaired? (choose the best answer)

- [ ] The person who broke the build.
- [ ] The next person who needs the build to complete successfully.
- [ ] The person assigned to the configuration management role within the team.
- [ ] The Tester responsible for validating builds.
- [ ] Whoever the Developers agree should fix it.

### What happens during Sprint 0? (choose the best answer)

- [ ] Establish base System Architecture and design, install version control and continuous integration setup.
- [ ] There is no such thing as Sprint 0.
- [ ] Base System Architecture and design.
- [ ] Overall planning, base System Architecture, base design, version control and continuous integration setup.
- [ ] Requirements gathering, version control setup, and continuous integration setup.

### Product Backlog Items are refined by: (choose the best answer)

- [ ] The Developers.
- [ ] The Product Owner.
- [ ] The Scrum Team.
- [ ] The Business Analyst and the Product Owner.

### Who is responsible for the System Architecture of a Product being developed using Scrum? (choose the best answer)

- [ ] The Architect chosen by the Scrum Team.
- [ ] The Developers.
- [ ] The Software Architect.
- [ ] The Corporate Architect.

### Which of the following are attributes of a bad bug report? (choose the best three answers)

- [ ] Simple and repeatable reproduction steps.
- [ ] Vague statements or untested assumptions.
- [ ] Generic titles.
- [ ] One bug per report.
- [ ] Assigning blame.

### Who writes tests in a Scrum Team? (choose the best answer)

- [ ] The Developers.
- [ ] Coders.
- [ ] The Scrum Master.
- [ ] Quality Assurance Specialists.

### When do the Developers participate in Product Backlog Refinement? (choose the best answer)

- [ ] Anytime during the Sprint.
- [ ] Never. It is the sole responsibility of the Product Owner to refine the Product Backlog.
- [ ] Only during Refinement sessions planned by the Product Owner.
- [ ] As Part 1 of Sprint Planning.

### Who should be present during Product Backlog Refinement? (choose the best answer)

- [ ] The integration architects from the release department.
- [ ] Only the most senior Developers.
- [ ] Anyone that the Scrum Team decides will be valuable during Refinement.
- [ ] The Stakeholders.
- [ ] The external Business Analysts that have prepared the functional details.

### True or False: Programmers and Testers should not be included in refining Product Backlog Items.

- [ ] True.
- [ ] False.

### Why are automated builds important? (choose the best answer)

- [ ] Without them you cannot tell if your code works.
- [ ] You are unable to check-in code without one.
- [ ] They are part of your done criteria.
- [ ] They provide rapid assurance that defects and configuration management issues have not been introduced.

### Upon what type of process control is Scrum based? (choose the best answer)

- [ ] Defined.
- [ ] Empirical.
- [ ] Complex.
- [ ] Hybrid.

### When might a Sprint be abnormally cancelled? (choose the best answer)

- [ ] When the Sprint Goal becomes obsolete.
- [ ] When the sales department has an important new opportunity.
- [ ] When the Developers feel that the work is too hard.
- [ ] When it becomes clear that not everything will be finished by the end of the Sprint.

### Who should know the most about the progress toward a business objective or a release, and be able to explain the alternatives most clearly? (choose the best answer)

- [ ] The Scrum Master.
- [ ] The Project Manager.
- [ ] The Product Owner.
- [ ] The Developers.

### When many Scrum Teams are working on a single Product, what best describes the Definition of Done? (choose the best answer)

- [ ] Each Scrum Team defines and uses its own. The differences are discussed and reconciled during a
hardening Sprint.
- [ ] The Scrum Masters from each Scrum Team define a common Definition of Done.
- [ ] Each Scrum Team uses its own, but must make their definition clear to all other teams so the differences are known.
- [ ] All Scrum Teams must have a Definition of Done that makes their combined Increment valuable and useful.

### During a Sprint, a Developer determines that the Scrum Team will not be able to complete the items in their forecast. Who should be present to review and adjust the Product Backlog Items selected? (choose the best answer)

- [ ] The Product Owner and the Developers.
- [ ] The Scrum Master, the Project Manager, and the Developers.
- [ ] The Product Owner and all Stakeholders.
- [ ] The Developers.

### When should the Developers on a Scrum Team be replaced? (choose the best answer)

- [ ] As needed, with no special allowance for changes in productivity.
- [ ] Never, it reduces productivity.
- [ ] As needed, while taking into account a short-term reduction in productivity.
- [ ] Every Sprint to promote shared learning.

### When is a Sprint over? (choose the best answer)

- [ ] When all Product Backlog Items meet their Definition of Done.
- [ ] When the timebox expires.
- [ ] When the Product Owner says it is done.
- [ ] When all the tasks are completed.

### When does the next Sprint begin? (choose the best answer)

- [ ] Immediately following the next Sprint Planning.
- [ ] Immediately after the conclusion of the previous Sprint.
- [ ] When the Product Owner is ready.
- [ ] Next Monday.

### What does it mean to say that an event has a timebox? (choose the best answer)

- [ ] The event must happen at a set time.
- [ ] The event can take no more than a maximum amount of time.
- [ ] The event must happen by a given time.
- [ ] The event must take at least a minimum amount of time.

### Who is required to attend the Daily Scrum? (choose the best answer)

- [ ] The Developers and Product Owner.
- [ ] The Developers.
- [ ] The Developers and Scrum Master.
- [ ] The Scrum Master and Product Owner.
- [ ] The Scrum Team.

### When does a Developer become accountable for the value of a Product Backlog Item selected for the Sprint? (choose the best answer)

- [ ] Whenever a team member can accommodate more work.
- [ ] At the Sprint Planning Event.
- [ ] During the Daily Scrum.
- [ ] Never. The entire Scrum Team is accountable for creating value every Sprint.

### Who is on the Scrum Team? (choose the best three answers)

- [ ] Project Manager.
- [ ] Developers.
- [ ] The Product Owner.
- [ ] The Scrum Master.

### Which of the following are examples of a Scrum Team practicing Scrum poorly or not exhibiting traits of a self-managing Scrum Team? (choose the best three answers)

- [ ] The Developers have all the skills they need to create a valuable, useful Increment.
- [ ] Stakeholders attend the Daily Scrum to check on the Scrum Team's progress.
- [ ] The Developers create their own Sprint Backlog, reflecting all work that is required to meet the Definition of Done.
- [ ] The Developers are collaboratively selecting their own work during the Sprint.
- [ ] The Developers invite external Stakeholders to the Sprint Planning to ask them how to turn a Product Backlog Item into an Increment via a complete and detailed Sprint Backlog.
- [ ] The Developers are working within the boundaries of their organizations functional description and nicely handing off work from analyst to Developer to Tester to integration.

### The timebox for the Sprint Review is: (choose the best answer)

- [ ] As long as needed.
- [ ] 2 hours.
- [ ] 4 hours for a monthly Sprint. For shorter Sprints it is usually shorter.
- [ ] 4 hours and longer as needed.
- [ ] 1 day.

### The timebox for the Sprint Planning event is? (choose the best answer)

- [ ] 8 hours for a monthly Sprint. For shorter Sprints it is usually shorter.
- [ ] 4 hours.
- [ ] Whenever it is done.
- [ ] Monthly.

### True or False: The purpose of a Sprint is to produce a valuable and useful Increment of working Product.

- [ ] True.
- [ ] False.

### An organization has decided to adopt Scrum, but Management wants to change the terminology to fit with terminology already used. What will likely happen if this is done? (choose the best answer)

- [ ] Without a new vocabulary as a reminder of the change, very little change may actually happen.
- [ ] The organization may not understand what has changed with Scrum and the benefits of Scrum may be lost.
- [ ] Management may feel less anxious.
- [ ] All of the above.

### Who creates the Definition of Done? (choose the best answer)

- [ ] The Scrum Team, in a collaborative effort where the result is the common denominator of all members' definitions.
- [ ] The Scrum Master since they are responsible for the productivity of the Developers.
- [ ] The Product Owner since they are responsible for the Product's success.
- [ ] If it is not an organizational standard, the Scrum Team must create a Definition of Done appropriate for the Product.

### Who is responsible for managing the progress of work during a Sprint? (choose the best answer)

- [ ] The Developers.
- [ ] The most junior member of the team.
- [ ] The Scrum Master.
- [ ] The Product Owner.

### Who has the final say on the order of the Product Backlog? (choose the best answer)

- [ ] The Product Owner.
- [ ] The Stakeholders.
- [ ] The Scrum Master.
- [ ] The CEO.
- [ ] The Developers.

### Which Scrum events are timeboxed? (choose the best three answers)

- [ ] Release Retrospective.
- [ ] Sprint Planning.
- [ ] Sprint Testing.
- [ ] Refinement.
- [ ] Sprint Retrospective.
- [ ] Sprint Review.
- [ ] Release Testing.

### A Scrum Team consists of the following: (choose the best three answers)

- [ ] Product Owner.
- [ ] Scrum Master.
- [ ] Customers.
- [ ] Users.
- [ ] Developers.

### When does a Developer become the sole owner of a Sprint Backlog item? (choose the best answer)

- [ ] Whenever a team member can accommodate more work.
- [ ] During the Daily Scrum.
- [ ] At the Sprint Planning event.
- [ ] Never. All Sprint Backlog items are "owned" by the Developers on the Scrum Team even though each item may be implemented by an individual Developer.

### What is the role of Management in Scrum? (choose the best answer)

- [ ] Support the Product Owner with insights and information into high value Product and system capabilities. Support the Scrum Master to cause organizational change that fosters empiricism, self-management, bottom-up intelligence, and intelligent Product delivery.
- [ ] Continually monitor staffing levels of the Scrum Team.
- [ ] Monitor the progress of the Developers on the Scrum Team.
- [ ] Identify and remove people that are not working hard enough.

### Why is the Daily Scrum held at the same time and same place? (choose the best answer)

- [ ] The consistency reduces complexity.
- [ ] The place can be named.
- [ ] The Product Owner demands it.
- [ ] Rooms are hard to book and this lets it be booked in advance.

### The length of a Sprint should be: (choose the best answer)

- [ ] Short enough to keep the business risk acceptable to the Product Owner.
- [ ] Short enough to be able to synchronize the development work with other business events.
- [ ] One month or less.
- [ ] All of the above.

### How much work must the Developers complete for each Product Backlog Item they select for a Sprint? (choose the best answer)

- [ ] Enough so that each Product Backlog Item they select meets the Definition of Done.
- [ ] Analysis, design, programming, testing and documentation.
- [ ] As much as it can fit into the Sprint.
- [ ] All development work and at least some testing.

### The CEO asks the Developers to add a "very important" item to a Sprint that is in progress. What should the Developers do? (choose the best answer)

- [ ] Add the item to the current Sprint and drop an item of equal size.
- [ ] Inform the other members of the Scrum Team so the team can decide what to do.
- [ ] Add the item to the next Sprint.
- [ ] Add the item to the current Sprint without any adjustments.

### What is the recommended size for a Scrum Team? (choose the best answer)

- [ ] Minimum of 7.
- [ ] 9.
- [ ] Typically 10 or fewer people.
- [ ] 7 plus or minus 2.

### Which statement best describes a Product Owner's responsibility? (choose the best answer)

- [ ] Keep Stakeholders from distracting the Developers.
- [ ] Optimizing the value of the work the Scrum Team does.
- [ ] Managing the project and ensuring that the work meets the commitments to the Stakeholders.
- [ ] Directing the Developers.

### True or False: 100% Code Coverage of tests guarantees bug-free Software.

- [ ] True.
- [ ] False.

### Who is responsible for monitoring the progress of work during a Sprint? Pick 1

- [ ] Product Owner.
- [ ] Project Manager.
- [ ] Scrum Master.
- [ ] Development Team.

### Who can tell the Developers what to work on?

- [ ] The Product Owner.
- [ ] The COO, CEO and CTO.
- [ ] The Project Manager.
- [ ] The Scrum Master.

### Who can do the work to ensure the Product Backlog is refined to a state that it is useful? Pick 2

- [ ] The Product Owner.
- [ ] The Business Analyst.
- [ ] Stakeholders.
- [ ] The Developers, with support from the Product Owner who is still accountable for it.

### SCRUM stands for:

- [ ] Super Creative Really Useful Method.
- [ ] Self organising Collaborative & Reflective Universal Method.
- [ ] Safe Controlled Response Under Management.
- [ ] Nothing. Its not an acronym.

### Who manages the Sprint Backlog?

- [ ] The Scrum Master.
- [ ] The Product Owner.
- [ ] The Developers.
- [ ] The Scrum Team.

### Which statement best describes a cross-functional team? Pick 1

- [ ] The team has a good mix of skills.
- [ ] All members of the team have all the skills required to create the Product.
- [ ] The team has all competencies needed to accomplish the work.

### Bugs discovered out of Sprint should be:

- [ ] Investigated and fixed immediately by the Developers.
- [ ] Reviewed with the Product Owner and added to the Product Backlog.

### When could a release to Production occur if a Scrum Team produced a Done increment multiple times each day? (select all that apply)

- [ ] Mid Sprint.
- [ ] Every Sprint.
- [ ] Multiple times per day.
- [ ] Every Day.

### True or False: The Definition of Done is a mandatory part of Scrum.

- [ ] False.
- [ ] True.

### Who may be best positioned to decide who will be the Scrum Master for a new Scrum Team?

- [ ] The Developers.
- [ ] The Product Owner.
- [ ] The Developers and the Product Owner.
- [ ] Stakeholders.

### True or False: A Scrum Team must be 10 or fewer people.

- [ ] True.
- [ ] False.

### Continuous Integration (CI) provides the following advantages: (select all that apply)

- [ ] Less back-tracking to discover where things went wrong, so you can spend more time building features.
- [ ] Allows teams to detect problems early.
- [ ] Automatically eliminates all integration issues.
- [ ] Stop waiting to find out if your code's going to work.

### Sizing is best when: (select all that apply)

- [ ] It is done by the actual Developers who will carry out the work.
- [ ] It is done by a group, rather than an individual.
- [ ] Estimates are made relative to similar things.
- [ ] It is done by the Lead Developer alone.

### Select the correct timebox for each Scrum event.

- [ ] Sprint Planning - 8 hours or less, Daily Scrum - 30 minutes or less, Sprint Review - 4 hours or less, Sprint Retrospective - 3 hours or less.
- [ ] Sprint Planning - 4 hours or less, Daily Scrum - 15 minutes or less, Sprint Review - 8 hours or less, Sprint Retrospective - 4 hours or less.
- [ ] Sprint Planning - 4 hours or less, Daily Scrum - 15 minutes or less, Sprint Review - 4 hours or less, Sprint Retrospective - 4 hours or less.
- [ ] Sprint Planning - 8 hours or less, Daily Scrum - 15 minutes or less, Sprint Review - 4 hours or less, Sprint Retrospective - 3 hours or less.

### True or False: Planning Poker must be used by Scrum Teams.

- [ ] False.
- [ ] True.

### What is Test Driven Development?

- [ ] Appointing the QA Tester as the Lead Developer.
- [ ] The practice of writing Unit Tests prior to writing the implementation code.
- [ ] Integrating Testers and Developers.
- [ ] The process of motivating Testers.

### True or False: Tasks in a Sprint Backlog must be estimated in days/hours.

- [ ] True.
- [ ] False.

### Who is responsible for testing in Scrum?

- [ ] The Product Owner.
- [ ] The Developers.
- [ ] Stakeholders.
- [ ] The Scrum Master.

### Bugs discovered in Sprint should be:

- [ ] Fixed immediately.
- [ ] Ignored if not created by me.
- [ ] Send to the Junior Developer to fix.
- [ ] Added to the bug tracking repository and left for a later Sprint.

### True or False: YAGNI states that most systems work best if they are kept simple rather than made complicated.

- [ ] False.
- [ ] True.

### Which one is a Scrum Value? (pick all that apply)

- [ ] Focus.
- [ ] Commitment.
- [ ] Courage.
- [ ] Creativity.
- [ ] Accountability.

### What is NOT the name of the original program in source control from which branches are taken?

- [ ] Mainline.
- [ ] Feature.
- [ ] Trunk.
- [ ] Master.

### What is the Static Analysis?

- [ ] Analysis performed on code in a non-runtime environment.
- [ ] Analysis performed on Software at runtime.
- [ ] Duties of a Business Analyst.
- [ ] Checking static classes' readability.

### Which one is true about Black-box testing? (pick all that apply)

- [ ] Tester can see inside the component or system under test.
- [ ] Tests that are done can be either functional or non-functional.
- [ ] The internal structure of the item being tested is known to the Tester.
- [ ] Tester cannot see inside the component or system under test.
- [ ] The internal structure of the item being tested is not known to the Tester.

### Which expression is NOT used for KISS principle?

- [ ] Keep it simple, silly.
- [ ] Keep it safe and same.
- [ ] Keep it small and simple.
- [ ] Keep it simple stupid.

### What shapes can be an Architecture Spike? (pick all that apply)

- [ ] A prototype which will be thrown away.
- [ ] A comprehensive architecture implementation.
- [ ] A quick and dirty implementation.
- [ ] A plan for all required architectures of the Product.

### What does cross-functionality mean in a Scrum Team?
- [ ] Each Developer who has a special skill and works with all Scrum Teams of the organization is cross-functional.
- [ ] The Development Team should collaborate with all other functional departments of the organization.
- [ ] The Development Team should have all required skills to create potentially releasable Increments without any dependency to others outside the team.
- [ ] Each Development Team member should have all required skills to create potentially releasable Increments without any dependency to others outside the team.

### What is the proper reaction of the Development Team in the middle of the Sprint when they found they have over committed?

- [ ] They should ask the Product Owner to cancel the Sprint.
- [ ] They extend the Sprint duration.
- [ ] They can add new team members to the team.
- [ ] They negotiate with the Product Owner about the scope of the Sprint Backlog Items.

### What is the Sprint Planning time-box for two-week Sprints?

- [ ] Usually 2 hours.
- [ ] Usually 8 hours.
- [ ] Usually 6 hours.
- [ ] Usually 4 hours.
